hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecord.java

```diff
@@ -115,7 +115,7 @@ public GenericRecordBuilder newBuilderWithClone() {
     public FieldKind getFieldKind(@Nonnull String fieldName) {
         FieldDescriptor field = schema.getField(fieldName);
         if (field == null) {
-            throw new IllegalArgumentException("Field name " + fieldName + " does not exist in the schema");
+            return FieldKind.NONE;
```

Comment:
?

Reply:
Renamed it to NOT_AVAILABLE and removed the hasField from the GenericRecord API as discussed